src/mo_interpreter/interpret.ml

```diff
@@ -995,12 +1021,33 @@ and interpret_func env name shared_pat pat f c v (k : V.value V.cont) =
 
 (* Programs *)
 
+let ensure_management_canister env =
+  if V.Env.mem "" (!(env.actor_env))
+  then ()
+  else
+    env.actor_env :=
+      V.Env.add
+        (* ManagementCanister with raw_rand (only) *)
```

Comment:
Can't you just initialize the actor env with the management canister preinstalled, instead adding it later? (If no, maybe leave a comment somewhere nearby?)

Also, maybe move the `V.Obj` literal into its own top-level definition?

Reply:
Yeah, I tried, but its awkward without losing the logging and access to other async helpers that Andreas provided.

For the ir interpreter, the initial env actually depends on the flavour so has to be done late.

This was of a rabbit hole than I had intended, though I guess it fixes other infelicities in the interpreter.